webpack/scenes/ContentViews/Delete/Steps/CVDeletionFinish.js

```diff
@@ -21,53 +23,59 @@ const CVDeletionFinish = () => {
     selectRemoveCVVersionResponse(state, cvId, cvEnvironments));
   const removeCVStatus = useSelector(state =>
     selectRemoveCVVersionStatus(state, cvId, cvEnvironments));
-  const removeResolved = removeCVStatus === STATUS.RESOLVED;
   const dispatch = useDispatch();
   const [removeDispatched, setRemoveDispatched] = useState(false);
-  const { push } = useHistory();
+  const { push, location: { pathname } } = useHistory();
 
   useDeepCompareEffect(() => {
-    if (removeResolved && removeDispatched) {
-      dispatch(getContentViews());
-      push('/content_views');
-      setIsOpen(false);
-    }
-    if (removeCVStatus === STATUS.ERROR) {
+    // If there is an error, we still want to refresh or redirect to show it.
+    if (removeDispatched && removeCVStatus === STATUS.ERROR) {
+      if (pathname === '/content_views') dispatch(getContentViews());
+      else push('/content_views');
       setIsOpen(false);
     }
-  }, [removeCVResponse, removeCVStatus, removeResolved,
-    setIsOpen, dispatch, cvId, removeDispatched, push]);
+  }, [dispatch, pathname, push, removeCVStatus, removeDispatched, setIsOpen]);
 
   useDeepCompareEffect(() => {
     if (!removeDispatched) {
       let params = {
         id: cvId,
+        destroy_content_view: true,
       };
 
       if (affectedActivationKeys) {
-        const activationKeysParams = {
+        params = {
+          ...params,
           key_content_view_id: selectedCVForAK,
           key_environment_id: selectedEnvForAK[0].id,
         };
-        params = { ...activationKeysParams, ...params };
       }
 
       if (affectedHosts) {
-        const hostParams = {
+        params = {
+          ...params,
           system_content_view_id: selectedCVForHosts,
           system_environment_id: selectedEnvForHost[0].id,
         };
-        params = { ...hostParams, ...params };
       }
 
-      const deletionParams = { destroy_content_view: true };
-      params = { ...deletionParams, ...params };
-      dispatch(removeContentViewVersion(cvId, cvId, cvEnvironments, params));
+      dispatch(removeContentViewVersion(
+        cvId, cvId, cvEnvironments, params,
+        // Only when we are on the contentViews page do we need to call getContentViews.
+        () => {
+          if (pathname === '/content_views') dispatch(getContentViews());
+          else push('/content_views');
+          setIsOpen(false);
+        },
+        () => {
+          setIsOpen(false);
```

Comment:
We do need to add this on publish/promote as well. Tested errors on those before planning and the wizard doesn't autoclose.

Reply:
will do, tests don't like this syntax by the way, so you may see alternatives (like in the removeFinish).